src/prefix_argument/argument_trait.rs

```diff
@@ -1,106 +1,60 @@
-//! Trait implemented for all types usable as prefix command parameters. This file also includes
-//! the auto-deref specialization emulation code to e.g. support more strings for bool parameters
-//! instead of the `FromStr` ones
+//! Trait implemented for all types usable as prefix command parameters.
+//!
+//! Many of these implementations defer to [`serenity::ArgumentConvert`].
 
 use super::{pop_string, InvalidBool, MissingAttachment, TooFewArguments};
 use crate::serenity_prelude as serenity;
-use std::marker::PhantomData;
 
-/// Full version of [`crate::PopArgument::pop_from`].
+/// The result of `<T as PopArgument>::pop_from`.
```

Comment:
```suggestion
/// The result of [`PopArgument::pop_from`].
```
Should keep the doc link.